Profile update

ZhiLinR opened the pull request:
updated profile microservice to include changes to the following (as per meeting)

1. User uid to use a separate field, autogenerated using ObjectId() method, sample in readme
2. tried out response and error handling middlewares + abit of standardising response formatting

also fixed a behavioural lapse in one of the endpoints - POST api/profile (register new user) - currently sensitive user data and profiling data are stored separately in collections, original post request did not trigger a record insertion into the profiling data collection

```
// user sensitive data
{
  "_id": {
    "$oid": "666350518e5c4522aed85893"
  },
  "uid": "666350518e5c4522aed85892",
  "email": "sammyho@example.com",
  "name": "Sammy"
}
```

```
// larger, more generic profiling data?
{
  "_id": {
    "$oid": "666350518e5c4522aed85894"
  },
  "uid": "666350518e5c4522aed85892",
  "lastModified": {
    "$date": "2024-06-07T18:26:12.711Z"
  },
  "gender": "female",
  "organisation_type": "banking",
  "title": "CFO"
}
```

cre8tion commented on microsvcs/user/README.md:
````diff
@@ -14,14 +27,36 @@ Updates specific **profiling** data (ie. organisation type, position etc) for a
+```
+{
+    "success": true,
+    "message": {
````
Dont return mongodb response as api response, just return your own custom response if needed

cre8tion commented on microsvcs/user/app/controller/profile_con.js:
```diff
@@ -7,33 +8,30 @@ const QUERIES = require("../model/profile.js")
  * @param {String} uid - unique uid field
  */
-exports.getProfileByUID = async (req, res) => {
+exports.getProfileByUID = async (req, res, next) => {
```
I believe next is only for error handling middleware, please verify on that

cre8tion replied:
okay seems like you have two middleware, maybe thats why. But try to verify on that

cre8tion commented on microsvcs/user/app/controller/profile_con.js:
```diff
@@ -7,33 +8,30 @@ const QUERIES = require("../model/profile.js")
-            res.status(404).send({ message: error.message });
+            throw new Error("No Reference Found")
         }
+    } catch (error) {
```
I think the error handling is supposed to catch automatically without you calling next

cre8tion reviewed (commented):
Looks okay, except im not very sure on the middleware portion

ZhiLinR commented:
changes:
1. removed MongoDB API Response
2. moved constants out of try blocks
3. standardised usage of next() in catch blocks as well
4. updated readme

cre8tion commented:
LGTM, feel free to merge
